collab2/foraging/toolkit/filtering.py

```diff
@@ -0,0 +1,35 @@
+from typing import Callable, List, Optional
+
+import numpy as np
+import pandas as pd
+
+
+# PP_comment : even with passing constraint params as a dictionary, we might have a limitation :
+# we cannot have a constraint that depends on quantities derived within the predictor function (eg, fastest v)
+# One workaround could be to locally (i.e. within the function) modify foragersDF with the derived quantities
+# and pass that to the constraint function via filter_by_distance
+def filter_by_distance(
+    foragersDF: pd.DataFrame,
+    f: int,
+    t: int,
+    interaction_length: float,
+    interaction_constraint: Optional[
+        Callable[[List[int], int, int, pd.DataFrame, Optional[dict]], List[int]]
+    ] = None,
+    interaction_constraint_params: Optional[dict] = None,
+) -> List[int]:
+    positions = foragersDF[foragersDF["time"] == t].copy()
```

Comment:
if `deepcopy()` resolved the issue in the `local_windows` PR, we should probably consistently do the same thing here as later you want to be adding stuff to windows, and a similar issue can arise, what do you think?

Reply:
ok sure!